### PR TITLE
Clear out master, move 0.10.x dev to the stable branch

### PR DESCRIPTION
**Pull request by bf4**
Status:

- ❗️ All existing PRs against master will need to be closed and re-opened against 0-10-stable, if so desired
- ❗️ Master, for the moment, won't have any released version of AMS on it.

Changes to 0.10.x maintenance:

- The 0.10.x version has become a huge maintenance version.  We had hoped to get it in shape for a 1.0 release, but it is clear that isn't going to happen.  Almost none of the maintainers from 0.8, 0.9, or earlier 0.10 are still working on AMS. We'll continue to maintain 0.10.x on the 0-10-stable branch, but maintainers won't otherwise be actively developing on it
  - We may choose to make a 0.11.x ( 0-11-stable) release based on 0-10-stable that just removes the deprecations.

Changes in this PR:

- The code and docs have been removed from master to signify that development against master has been suspended.  When choosing ways that people might be surprised by the current state of 0.10.x development, we thought it would be simpler to make master simply incompatible with 0.10.x PRs than to ensure none are merged.

What's happening to AMS:

- There's been a lot of churn around AMS since it began back in Rails 3.2 and a lot of new libraries are around and the JSON:API spec has reached 1.0.
- If there is to be a 1.0 release of AMS, it will need to address the general needs of serialization in much the way ActiveJob can be used with different workers.
- The next major release *is* in development. I am building and extracting it from an app I'm working on and thereby dogfooding it myself to ensure it to see if it makes sense.  I'm starting simple and avoiding, at least at the outset, all the complications in AMS version, especially all the implicit behavior from guessing the serializer, to the association's serializer, to the serialization type, etc.
- The basic idea is that models to serializers are a one to many relationship.  Everything will need to be explicit.  If you want to serialize a User with a UserSerializer, you'll need to call it directly.  The serializer will essentially be for defining a basic JSON:API resource object: id, type, attributes, and relationships. The serializer will have an as_json method and can be told which fields (attributes/relationships) to serialize to JSON and will likely *not* know serialize any more than the relations id and type.  Serializing anything more about the relations would require code that called a serializer. (This is still somewhat in discussion).
- If this works out, the idea is to get something into Rails that existing libraries can use.

Steps:

1. review this PR, clearing out master, moving 0-10 dev to 0-10-stable
1. merge it
1. make a PR from dev to master and work on it
 - add rails rubocop https://github.com/rails/rails/search?utf8=%E2%9C%93&q=rubocop&type=
 - I'm currently using the dev branch (in which the gem is named `ams` in our new api I'm building, so am dogfooding it)

### CODE REVIEW

File: CHANGELOG-0-08.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
+## 0.08.x
```
Comment (bf4) [inline CHANGELOG-0-08.md]: gets its own changelog

File: CHANGELOG-0-09.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
+## 0.09.x
```
Comment (bf4) [inline CHANGELOG-0-09.md]: gets its own changelog

File: CHANGELOG-0-10.md [hunk 3]
```diff
@@ -0,0 +1,466 @@
+## 0.10.x
```
Comment (bf4) [inline CHANGELOG-0-10.md]: gets its own changelog

File: CHANGELOG.md [hunk 4]
```diff
@@ -10,641 +10,10 @@ Fixes:
 
 Misc:
 
-### [v0.10.6 (2017-05-01)](https://github.com/rails-api/active_model_serializers/compare/v0.10.5...v0.10.6)
+## [0.10.x](CHANGELOG-0-10.md)
```
Comment (bf4) [inline CHANGELOG.md]: just references other changelogs

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -4,13 +4,7 @@ Before opening an issue, try the following:
...
 ##### Consult the documentation
 
-See if your issue can be resolved by information in the documentation.
-
-- [0.10 (master) Documentation](https://github.com/rails-api/active_model_serializers/tree/master/docs)
-  - [![API Docs](http://img.shields.io/badge/yard-docs-blue.svg)](http://www.rubydoc.info/github/rails-api/active_model_serializers/v0.10.0)
-  - [Guides](docs)
-- [0.9 (0-9-stable) Documentation](https://github.com/rails-api/active_model_serializers/tree/0-9-stable)
-- [0.8 (0-8-stable) Documentation](https://github.com/rails-api/active_model_serializers/tree/0-8-stable)
+See if your issue can be resolved by information in the [documentation](README.md).
```
Comment (bf4) [inline CONTRIBUTING.md]: these links weren't up to date

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -43,7 +37,9 @@ for discussion or add your comments to existing ones.
 We also gladly welcome pull requests. When preparing to work on pull request,
 please adhere to these standards:
 
-- Base work on the master branch unless fixing an issue with
+- Base work on the relevant branch:
+  [0.10-stable](https://github.com/rails-api/active_model_serializers/tree/0-10-stable)
```
Comment (bf4) [inline CONTRIBUTING.md]: 0-10 is no longer master

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -52,10 +48,10 @@ please adhere to these standards:
 - Note any specific areas that should be reviewed.
 - Include tests.
 - The test suite must pass on [supported Ruby versions](.travis.yml)
-- Include updates to the [documentation](https://github.com/rails-api/active_model_serializers/tree/master/docs)
+- Include updates to the [documentation](docs)
```
Comment (bf4) [inline CONTRIBUTING.md]: this won't work on master.. maybe should link to 0-10? or no link?

File: README.md [hunk 8]
```diff
@@ -86,10 +20,11 @@ Thanks!
 ## Documentation
 
 If you're reading this at https://github.com/rails-api/active_model_serializers you are
-reading documentation for our `master`, which may include features that have not
-been released yet. Please see below for the documentation relevant to you.
+reading documentation for our `master`, which is not yet released.
```
Comment (bf4) [inline README.md]: 0-10 is no longer master

File: README.md [hunk 9]
```diff
@@ -86,10 +20,11 @@ Thanks!
...
+
+Please see below for the documentation relevant to you.
 
-- [0.10 (master) Documentation](https://github.com/rails-api/active_model_serializers/tree/master)
+- [0.10 (0-10-stable) Documentation](https://github.com/rails-api/active_model_serializers/tree/0-10-stable)
```
Comment (bf4) [inline README.md]: 0-10 is no longer master

File: active_model_serializers.gemspec [hunk 10]
```diff
@@ -1,11 +1,8 @@
 # coding: utf-8
-lib = File.expand_path('../lib', __FILE__)
-$LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
-require 'active_model/serializer/version'
 
 Gem::Specification.new do |spec|
   spec.name          = 'active_model_serializers'
-  spec.version       = ActiveModel::Serializer::VERSION
+  spec.version       = "1.0.0-dev"
```
Comment (bf4) [inline active_model_serializers.gemspec]: just picked something so tests can run

**Comment by mauriciopasquier** [comment]
What about Adapters? I'm in the process of migrating from 0.8 to 0.10, and was planing to write a few Adapters to fit my needs. Right now (0.8) I'm basically using model Serializers to describe a "neutral" hash of the model and I use specific serializers to convert that neutral representation to GeoJSON, CSV and JSON:API.

Will the Serializer/Adapter model be compatible with the new work beign done? From what I read here, my approach of having a specific serializer for a needed format (i.e. GeoJSON) will be prefered.

**Comment by bf4** [comment]
@mauriciopasquier At this time, it appears that adapters may be a special case for serializers.  I haven't fully fleshed this out, yet, but basically you'd have your UserSerializer and your DocumentSerializer and combine the two to serialize the user into a JSONAPI response document. I'm starting from a place with no magic so that the fundamentals of how the combine the parts are less hard to work with.  I imagine something like DocumentSeriarlizer.new(UserSerializer.new(user)).as_json.

**Comment by mauriciopasquier** [comment]
@bf4 So I guess I'll stay with that approach. What I'm doing exactly in 0.8 is:

    # In a controller
    format.geojson do
      render json: @users, serializer: GeojsonCollectionSerializer
    end

    # In the collection serializer `as_json` method
    object.map do |user|
      GeojsonUserSerializer.new(user).as_json
    end

Of course this implementation makes one serializer depends on the other, but it works for my case. Will 1.0.0 provide this `DocumentSerializer` (or similar) as 0.10 does with a the JSON:API Adapter?

Thanks for answering :)

**Comment by bf4** [comment]
> Will 1.0.0 provide this DocumentSerializer (or similar) as 0.10 does with a the JSON:API Adapter?

probably... like I said, I'm building and extracting this out of our code, while talking to stakeholders in Rails and API-gems...

**Comment by thomasnal** [comment]
I have come to re-check some details in documentation and spotted the ideas at the top planned for AMS.  

AMS is instrumental to our project, we use it heavily to keep sane view on serialization and keep the team producing clean code.

Some of the ideas are related to removing functionality that works perfectly currently therefore it is worry-some.  I came to this PR to see background behind the decisions that I have not found.  Is there other source where I can find rationals behind the decisions?  Especially for removing that what works now?

> DocumentSeriarlizer.new(UserSerializer.new(user)).as_json

This snippet looks backwards from what AMS does currently.  However, perhaps it means that below will keep working?  Can you please shed more light @bf4?
```ruby
class Profile:UsersShowSerializer < ActiveModel::Serializer
  attributes ...
end

class User::UsersShowSerializer < ActiveModel::Serializer
  attribute :profile, :serializer => Profile::UsersShowSerializer
  ...
end

class CurrentUsersController < ApplicationController
  def show
    render json: user, status: :ok
  end
end
```

**Pull request merged**